delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/indexing/BlazegraphCoordinator.scala

```diff
@@ -45,26 +45,38 @@ final class BlazegraphCoordinator private (
   def run(offset: Offset): Stream[Task, Elem[Unit]] = {
     fetchViews(offset).evalMap { elem =>
       elem
-        .traverse {
-          case active: ActiveViewDef =>
-            IndexingViewDef
-              .compile(
-                active,
-                compilePipeChain,
-                graphStream,
-                sink(active)
-              )
-              .flatMap { projection =>
-                cleanupCurrent(active.ref) >>
-                  supervisor.run(
-                    projection,
-                    for {
-                      _ <- createNamespace(active)
-                      _ <- cache.put(active.ref, active)
-                    } yield ()
+        .traverse { v =>
+          cache.get(v.ref).flatMap { cachedView =>
+            (cachedView, v) match {
+              case (Some(cached), active: ActiveViewDef) if cached.projection == active.projection =>
+                for {
+                  _ <- cache.put(active.ref, active)
+                  _ <- Task.delay {
+                         logger.info(s"Index ${active.projection} already exists and will not be recreated.")
+                       }
+                } yield ()
+              case (_, active: ActiveViewDef)                                                      =>
+                IndexingViewDef
```

Comment:
Maybe create a private method compile here to improve a little readibility ?
Something like:
```scala
private def compile(v: ActiveViewDef) =
    IndexingViewDef.compile(v, compilePipeChain, graphStream, sink(v))
```